Apollo & React update, add suspense support

YoruNoHikage opened the pull request.

YoruNoHikage committed:
Update apollo, remove react-apollo-hooks, add suspense tmp solution

YoruNoHikage committed:
Keep queries into a module variable

YoruNoHikage committed:
Add observable to result to be able to throw the right promise

YoruNoHikage committed:
Update graphql

YoruNoHikage committed:
Update storybook to remove old react version from deps

FranzPoize commented:
Focus task not updating dashboard:
- Delete all projects
- Create a new project from a template
- go to dashboard
- try to focus a task through drag and drop

(This happens on the first visit to dashboard after creating the project not the subsequent)

FranzPoize commented:
Could you commit without the localhost:4000 backend :)

YoruNoHikage committed:
Update react-scripts, babel and eslint

YoruNoHikage committed:
Updating gitignore for envrc

YoruNoHikage committed:
Reuse the previous forceUpdate callbacks to avoid losing some hooks

Let me talk about this one since I spent a few days tracking it down.
Sometimes, we make the same query in a child component, don't ask me why
I don't know. I don't know anything about this, trust me. Anyway, since
the query is exactly the same, the trick I used by keeping everything in the
module - since, let me repeat, the hooks are dropped when throwing something -
makes the whole thing not updating the simultaneous queries and it's annoying.
The solution that this commit introduces is the dirty way to make everything
works by just stacking all the forceUpdate callbacks into the new one. And that
works - with some memory leaks maybe, I don't know, who knows...

FranzPoize commented on src/utils/useApolloQuery.js:
```diff
@@ -0,0 +1,81 @@
+	const context = useContext(getApolloContext());
+	const [tick, forceUpdate] = useReducer(x => x + 1, 0);
+	const updatedOptions = options
+		? Object.assign(Object.assign({}, options), {query})
```
```suggestion
		? {...options, {query: ...query}}
```
ca revient au même si je me trompe pas.

FranzPoize replied:
```suggestion
		? {...options, query}
```
Si la ref a la query peut être gardé.

FranzPoize commented on src/utils/useApolloQuery.js:
```diff
@@ -0,0 +1,81 @@
+
+	const previousForceUpdate = queryData.forceUpdate;
+
+	if (queryData.forceUpdate !== forceUpdate) {
```
```suggestion
	if (previousForceUpdate !== forceUpdate) {
```
Ptet plus lisible

YoruNoHikage committed:
Apply suggestions, a little bit clearer

Co-Authored-By: Francois Poizat <francois.poizat@gmail.com>
